### PR TITLE
Update for v3

### PR DESCRIPTION
**Pull request by joextodd**
**Who:** Me
**Why:** For release to Arduino Library Manager
**What:** 

- Replaced with necessary code for uploading to Arduino Library.
- Added example for ESP32 + SPH0645 + UDA1334A

Requires: Latest engine memory improvements, testing with other protocols. Currently only works with 'embedded' protocol.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,84 +1,143 @@
...
 
-## Audio output
+    err = chirp_connect_set_callback_ptr(connect, connect);
+    if (err != CHIRP_CONNECT_OK)
+        return;
 
-On the Uno, the main sound signal is supplied on digital pin 3, and on the Mega it's on digital pin 9. Use pin I/O_0 on the Engduino. Connect this pin and a ground pin to your compatible speaker or audio device.
+    err = chirp_connect_start(connect);
+    if (err != CHIRP_CONNECT_OK)
+        return;
```
Comment (LittleMonsterFr) [inline README.md]: Users will need to set the sample rate if it's not 44.1 kHz

File: README.md [hunk 2]
```diff
@@ -1,84 +1,143 @@
...
+## Callbacks
 
-Although amplification & smoothing circuitry would be ideal, if you want to keep things simple you can attach a small speaker or an earbud directly to the Arduino pins. The exact nature of the smoothing, and so the fine characteristics of the sound, will depend on the particular hardware that you attach. We have had success directly connecting the pins to a 3.5mm audio lead and plugging this into a device such as an old radio. Please be aware that such a signal won't match the official audio jack electrical specifications (so keep the volume low), and you will certainly be overdriving an earbud if you connect it this way (don’t put it in your ear!). Whatever hardware you connect to the Arduino you do at your own risk; it's your responsibility to check for compatibility.
+The received data is passed back to the `onReceivedCallback` function. If the payload pointer is null then there has been an error decoding the data.
 
-#### Pulse Width Modulation
+    void
+    onReceivedCallback(void *connect, uint8_t *payload, size_t length, uint8_t channel) {
+        if (payload) {
+            char *hexString = chirp_connect_as_string(connect, payload, length);
```
Comment (LittleMonsterFr) [inline README.md]: `hexString ` needs to be freed by `chirp_connect_free` before leaving the callback

File: examples/receive.ino [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+onReceivingCallback(void *connect, uint8_t *payload, size_t length, uint8_t channel)
+{
+  Serial.println("Receiving data...");
+}
+
+void
+onReceivedCallback(void *connect, uint8_t *payload, size_t length, uint8_t channel)
+{
+  if (payload) {
+    char *hexString = chirp_connect_as_string((chirp_connect_t *)connect, payload, length);
```
Comment (LittleMonsterFr) [inline examples/receive.ino]: Same for freeing `hexString`

**Comment by joextodd** [comment]
I think it may as well stay public yes. There shouldn't be too much work to do to publish the library. 

We may as well wait for your optimisations before publishing though. Also need to test which configs can fit in memory following this, and clearly document this in the README - currently this only works with my custom `embedded` config.

**Comment by LittleMonsterFr** [comment]
Okay, can we close the PR for the moment and re-open it later ?

**Pull request closed without merging**